PyPDF2/_page.py

```diff
@@ -1069,48 +1075,188 @@ def extract_text(self, Tj_sep: str = "", TJ_sep: str = "") -> str:
+                    .get_data()
+                    .decode("utf-8")
+                )
+                for l in (
+                    cm.strip()
+                    .replace("<", " ")
+                    .replace(">", "")
+                    .replace("[", " [ ")
+                    .replace("]", " ] ")
+                    .split("\n")
+                ):
+                    if l == "":
+                        continue
+                    if "beginbfrange" in l:
+                        processRg = True
+                    elif "endbfrange" in l:
+                        processRg = False
+                    elif "beginbfchar" in l:
+                        processChar = True
+                    elif "endbfchar" in l:
+                        processChar = False
+                    elif processRg:
+                        lst = [x for x in l.split(" ") if x]
+                        a = int(lst[0], 16)
+                        b = int(lst[1], 16)
+                        if lst[2] == "[":
+                            # lst = lst[3:].trim(' []').split(' ')
+                            for sq in lst[3:]:
+                                if "]":
+                                    break
+                                mapDict[a] = unhexlify(sq).decode("utf-16-be")
+                                a += 1
+                                assert a > b
+                        else:
+                            c = int(lst[2], 16)
+                            fmt = b"%%0%dX" % len(lst[2])
+                            while a <= b:
+                                mapDict[a] = unhexlify(fmt % c).decode("utf-16-be")
+                                a += 1
+                                c += 1
+                    elif processChar:
+                        lst = [x for x in l.split(" ") if x]
+                        a = int(lst[0], 16)
+                        mapDict[a] = unhexlify(lst[1]).decode("utf-16-be")
+            return fontType, dict(zip(range(256), encoding)), "".maketrans(mapDict)
+        # ------- end of buildCharmap ------
+        text: str = ""
+        output: str = ""
+        cmaps: Dict[str, Tuple[str, Dict[int,str],Dict[str,str]]]={}
+        if "/Font" in cast(DictionaryObject,self["/Resources"]):
+            for f in cast(DictionaryObject,cast(DictionaryObject,self["/Resources"])["/Font"]):
+                cmaps[f] = buildCharMap(f)
+        cmap: Union[str, Dict[int,str]]= {}
         content = self[PG.CONTENTS].get_object()
         if not isinstance(content, ContentStream):
-            content = ContentStream(content, self.pdf)
+            content = ContentStream(content, self.pdf, "charmap")
         # Note: we check all strings are TextStringObjects.  ByteStringObjects
         # are strings where the byte->string encoding was unknown, so adding
         # them to the text here would be gibberish.
+
+        # charSize = 1.0
+        # charScale = 0.0
+        spaceScale = 1.0
```

Comment:
Nitpic: I try to get [PEP8](https://peps.python.org/pep-0008/) compliant code. This makes it easier to read as it's one style.

I would still merge any PR that violates PEP8, but I would later clean those lines up.